project/profiles/capacity4more/modules/c4m/features/c4m_features_homepage/c4m_features_homepage.module

```diff
@@ -54,30 +54,54 @@ function c4m_features_homepage_c4m_suggested_groups_get_groups() {
       ->propertyCondition('nid', $user_groups['node'], 'NOT IN');
   }
 
+  $link_filters = array();
   if (!empty($interests)) {
     $query->fieldCondition('c4m_vocab_topic', 'tid', $interests, 'IN');
+
+    // Adds each interest to the exposed filters that will be pass to the
+    // groups page.
+    foreach ($interests as $key => $interest) {
+      $link_filters['f[' . $key . ']'] = 'c4m_vocab_topic:' . $interest;
+    }
   }
 
   $result = $query
     ->addTag('random')
-    ->range(0, 3)
+    ->range(0, 5)
     ->execute();
 
   if (empty($result['node'])) {
     return;
   }
 
+  // After selecting 5 random groups, they are ordered alphabetically.
+  $nodes = array();
   foreach ($result['node'] as $group) {
-    $node_view = node_view(node_load($group->nid), 'block_list');
+    $nodes[] = node_load($group->nid);
+  }
+  uasort($nodes, 'c4m_features_homepage_node_sort_by_title');
```

Comment:
The query is already sorted as random. The MySQL sort acts before the query limit. In our case we need to sort first as random, limit the amount of items, sort the selected items alphabetically. This is not possible to do it in MySQL without a subquery or with two selects. In this case, I prefer to do the sort on the PHP side.

On the other hand, we could drop the random sort, but then, we will always return the same items.